app/services/weights_and_measures.rb

```diff
@@ -45,9 +45,16 @@ def self.available_units
 
   def self.available_units_sorted
     self::UNITS.transform_values do |measurement_info|
+      # Filter to only include available units
       measurement_info.filter do |_scale, unit_info|
         available_units.include?(unit_info['name'])
-      end.sort.to_h # sort by unit (hash key)
+      end.
+        # Remove duplicates by name
+        uniq do |_scale, unit_info|
+        unit_info['name']
+      end.
+        # Sort by unit number
+        sort.to_h
```

Comment:
If we remove duplicates by name then, I guess, it randomly selects one of the duplicate options. So we may get the low precision entry or the high precision entry. We haven't actually discussed which version we want to use.